### PR TITLE
Google auth redirect

### PR DESCRIPTION
**Pull request by pallenSL**

### CODE REVIEW

File: Source/Auth.Web/Startup.fs [hunk 1]
```diff
@@ -1,15 +1,31 @@
...
 open IdentityServer3.Core.Models;
 open IdentityServer3.Core.Configuration
 open IdentityServer3.Core.Services.InMemory
 
 open Owin
+open Owin.Security.AesDataProtectorProvider
+open Microsoft.Owin.Security.Google
+
+
+let ConfigureIdentityProviders (app: IAppBuilder) (signIsAsType: string) =
```
Comment (csscottlogic) [inline Source/Auth.Web/Startup.fs]: By convention, functions in a module start with a lower case letter.

File: Source/Auth.Web/Startup.fs [hunk 2]
```diff
@@ -1,15 +1,31 @@
...
+  app.UseAesDataProtectorProvider()
+
+  let google = 
+    new GoogleOAuth2AuthenticationOptions(
+      SignInAsAuthenticationType = signIsAsType,
+      ClientId = "143169614954-i44rbgum14ledrhsk6nn4qn10ehsakd5.apps.googleusercontent.com",
+      ClientSecret = "AlT3XLziFyhDqdW41S256Pq5")
```
Comment (csscottlogic) [inline Source/Auth.Web/Startup.fs]: Do we want this publicly visible in source control?

File: Source/Auth.Web/Startup.fs [hunk 3]
```diff
@@ -1,15 +1,31 @@
...
+
+  app.UseGoogleAuthentication(google) |> ignore
```
Comment (csscottlogic) [inline Source/Auth.Web/Startup.fs]: Is the return value here useful, or do we need to just call this for its side effects?

File: Source/Auth.Web/Startup.fs (hunk 2 above)
Comment (jebbinSCL) [inline Source/Auth.Web/Startup.fs]: I think you'll have to recreate the client secret

File: Source/Auth.Web/Startup.fs (hunk 1 above)
Comment (pallenSL) [inline Source/Auth.Web/Startup.fs]: Ok

File: Source/Auth.Web/Startup.fs (hunk 2 above)
Comment (pallenSL) [inline Source/Auth.Web/Startup.fs]: App.config is currently in source control too...

File: Source/Auth.Web/Startup.fs (hunk 3 above)
Comment (pallenSL) [inline Source/Auth.Web/Startup.fs]: Which return value - the middleware insertion?  That has to be void (IAppBuilder, string)

File: Source/Auth.Web/Startup.fs (hunk 2 above)
Comment (jebbinSCL) [inline Source/Auth.Web/Startup.fs]: You can have the app.config reference another config file that we don't check in. If you want to do this, you can take a look at the TrelloSync project under infrastructure, or let me know and I can show you want you need to do.

Comment (pallenSL) [inline Source/Auth.Web/Startup.fs]: Will check TrelloSync for reading config.  Thanks!

**Comment by ianlovell** [comment]
Think given the urgency of getting a demo together, the client secrets can stay in code, and we can change them later

**Comment by pallenSL** [comment]
It's okay - done that.  A new PR though as I had to merge the structural changes.  One moment.

**Pull request closed without merging**